nflow-engine/src/main/java/io/nflow/engine/workflow/instance/WorkflowInstance.java

```diff
@@ -148,6 +148,11 @@
    */
   public final Optional<Integer> signal;
 
+  /**
+   * True if this instance is archived. Archived instances are read-only, no modifications are allowed to them.
```

Comment:
Are we enforcing read-only somehow?

Reply:
Yeah. None of the normal DAO methods support writing/updating archive tables. I have only added explicit archive table support to the single query and search operations.